packages/template-retail-react-app/app/pages/login/index.jsx

```diff
@@ -27,19 +35,57 @@ const Login = () => {
     const form = useForm()
     const location = useLocation()
     const einstein = useEinstein()
-    const {isRegistered} = useCustomerType()
+    const {isRegistered, customerType} = useCustomerType()
+    const customerId = useCustomerId()
     const login = useAuthHelper(AuthHelpers.LoginRegisteredUserB2C)
 
+    /************** merge basket for recurring users ***/
+    const prevAuthType = useRef()
```

Comment:
How about we pass in an initial value like this?

```suggestion
    const prevAuthType = useRef(customerType)
```